src/api/routes/bns/pricing.ts

```diff
@@ -0,0 +1,138 @@
+import * as express from 'express';
+import { RouterWithAsync, addAsync } from '@awaitjs/express';
+import { DataStore, DbBnsNamespace } from '../../../datastore/common';
+import {
+  makeRandomPrivKey,
+  getAddressFromPrivateKey,
+  TransactionVersion,
+  ReadOnlyFunctionOptions,
+  bufferCVFromString,
+  callReadOnlyFunction,
+  ClarityType,
+  tupleCV,
+  uintCV,
+  listCV,
+  ChainID,
+} from '@stacks/transactions';
+import { GetStacksNetwork, getBnsContractID } from './../../../bns-helpers';
+import {
+  BnsGetNamePriceResponse,
+  BnsGetNamespacePriceResponse,
+} from '@blockstack/stacks-blockchain-api-types';
+import { isValidPrincipal, logger } from './../../../helpers';
+
+export function createBnsPriceRouter(db: DataStore, chainId: ChainID): RouterWithAsync {
+  const router = addAsync(express.Router());
+  const stacksNetwork = GetStacksNetwork(chainId);
+
+  router.getAsync('/namespaces/:namespace', async (req, res) => {
+    const { namespace } = req.params;
+    if (namespace.length > 20) {
+      res.status(400).json({ error: 'Invalid namespace' });
+      return;
+    }
+    const randomPrivKey = makeRandomPrivKey();
+    const address = getAddressFromPrivateKey(
+      randomPrivKey.data,
+      chainId === ChainID.Mainnet ? TransactionVersion.Mainnet : TransactionVersion.Testnet
+    );
+    const bnsContractIdentifier = getBnsContractID(chainId);
+    if (!bnsContractIdentifier || !isValidPrincipal(bnsContractIdentifier)) {
+      logger.error('BNS contract ID not properly configured');
+      return res.status(500).json({ error: 'BNS contract ID not properly configured' });
+    }
+
+    const txOptions: ReadOnlyFunctionOptions = {
+      senderAddress: address,
+      contractAddress: bnsContractIdentifier,
+      contractName: 'bns',
+      functionName: 'compute-namespace-price?',
```

Comment:
This function no longer exists, we should be using `get-namespace-price` instead.

Reply:
Fixed in 291b74c